### PR TITLE
Better unique collection matchers

### PR DESCRIPTION
**Pull request by sksamuel**

### CODE REVIEW

File: documentation/docs/assertions/collections.md [hunk 1]
```diff
@@ -0,0 +1,50 @@
+---
+id: collections
+title: Collection Matchers
+slug: collection-matchers.html
+sidebar_label: Collections
+---
+
+This page describes the rich assertions (matchers) that are available for Collection, Iterable and Array types.
+
```
Comment (Kantis) [inline documentation/docs/assertions/collections.md:9]: Would be worthwhile to mention inspectors here as well IMO

File: kotest-assertions/kotest-assertions-core/src/commonMain/kotlin/io/kotest/matchers/collections/unique.kt [hunk 2]
```diff
@@ -36,10 +65,30 @@ fun <T> Collection<T>.shouldNotBeUnique(): Collection<T> {
...
+   override fun test(value: Collection<T>): MatcherResult {
+
+      val list = value.toMutableList()
+      list.toSet().forEach { list.remove(it) }
+      val duplicates = list.toList().distinct()
+
+      return MatcherResult(
+         duplicates.isEmpty(),
+         { "Collection should be unique but contained duplicates of ${duplicates.joinToString(", ")}" },
+         { "Collection should contain at least one duplicate element" })
```
Comment (Kantis) [inline kotest-assertions/kotest-assertions-core/src/commonMain/kotlin/io/kotest/matchers/collections/unique.kt:77]: Nice 👍🏼

File: kotest-assertions/kotest-assertions-core/src/jvmTest/kotlin/com/sksamuel/kotest/matchers/collections/UniqueTest.kt [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+import io.kotest.matchers.collections.shouldBeUnique
+import io.kotest.matchers.collections.shouldNotBeUnique
+
+class UniqueTest : FunSpec() {
+   init {
+
+      test("List.shouldBeUnique") {
+         shouldThrowWithMessage<AssertionError>("Collection should be unique but contained duplicates of 1") {
+            listOf(1, 1, 2).shouldBeUnique()
+         }
```
Comment (Kantis) [inline kotest-assertions/kotest-assertions-core/src/jvmTest/kotlin/com/sksamuel/kotest/matchers/collections/UniqueTest.kt:14]: Should we assert the message as well?

**Review by Kantis — APPROVED** [review]

File: documentation/docs/assertions/collections.md (hunk 1 above)
Comment (sksamuel) [reply documentation/docs/assertions/collections.md:9]: done

File: kotest-assertions/kotest-assertions-core/src/jvmTest/kotlin/com/sksamuel/kotest/matchers/collections/UniqueTest.kt (hunk 3 above)
Comment (sksamuel) [reply kotest-assertions/kotest-assertions-core/src/jvmTest/kotlin/com/sksamuel/kotest/matchers/collections/UniqueTest.kt:14]: We are, the message is the parameter to the assertion.

Comment (Kantis) [reply kotest-assertions/kotest-assertions-core/src/jvmTest/kotlin/com/sksamuel/kotest/matchers/collections/UniqueTest.kt:14]: Doh, didn't realize. I'm more used to this pattern :) 

```kotlin
shouldFail {
  listOf(1,1,2).shouldBeUnique()
}.message shouldBe """
  failure message..
"""
```

**Pull request merged**